### PR TITLE
Fixed profilecard making it dynamic

### PR DESCRIPTION
**Pull request by Lanthoiba2022**
## Summary

Fixes the ProfieCard Making the Status/Button dynamic. Eventually the BE API endpoint need to be updated to sync with the currently login user only after authentication so as to see his/her progress.

### CODE REVIEW

File: frontend/components/radialProfileCard.tsx [hunk 1]
```diff
@@ -15,54 +16,118 @@ import {
...
-
-import { HoverCard, HoverCardContent, HoverCardTrigger } from '@/components/ui/hover-card';
-
+} from "@/components/ui/dialog";
 import { Button } from '@/components/ui/button';
 import Image from 'next/image';
 
+// get API endpoint BE
+export const fetchUserData = async () => {
+  try {
```
Comment (krishnaacharyaa) [inline frontend/components/radialProfileCard.tsx:25]: This should be removed? It is used in utils profileCompletionCheck.ts I see?

File: frontend/components/radialProfileCard.tsx [hunk 2]
```diff
@@ -15,54 +16,118 @@ import {
...
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error('Error fetching user data:', error);
+    return null; 
+  }
+};
+
+export const calculateProfileCompletion = (userData: any) => {
+  let completedFields = 0;
```
Comment (krishnaacharyaa) [inline frontend/components/radialProfileCard.tsx:39]: Same

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]

File: frontend/components/radialProfileCard.tsx (hunk 1 above)
Comment (Lanthoiba2022) [reply frontend/components/radialProfileCard.tsx:25]: ohh profileCompletionCheck.ts  is a redundant file it's not needed I forgot to delete that file

Comment (krishnaacharyaa) [reply frontend/components/radialProfileCard.tsx:25]: No have it there, remove this @Lanthoiba2022

Comment (Lanthoiba2022) [reply frontend/components/radialProfileCard.tsx:25]: ohh Ok I'll make the changes

**Comment by Lanthoiba2022** [comment]
Please see if it's ok now

**Review by krishnaacharyaa — APPROVED** [review]
LGTM, thank you

**Pull request merged**